File: app/src/main/java/com/airbnb/HeroFragment.java
package com.airbnb;

import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.ButterKnife;
import butterknife.InjectView;

public class HeroFragment extends Fragment {
    public static final String ARG_HERO_ID = "hero_id";

    @InjectView(R.id.text) public TextView mTextView;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.hero_fragment, container, false);
        ButterKnife.inject(this, view);

        if (getArguments() == null || !getArguments().containsKey(ARG_HERO_ID)) {
            throw new IllegalArgumentException("You must specify a hero id!");
        }

        mTextView.setText("Hero item id " + getArguments().getLong(ARG_HERO_ID));

        return view;
    }
}

File: app/src/main/java/com/airbnb/HeroFragment.kt
package com.airbnb

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import kotlinx.android.synthetic.hero_fragment.*

public class HeroFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (getArguments() == null || !getArguments().containsKey(ARG_HERO_ID)) {
            throw IllegalArgumentException("You must specify a hero id!")
        }
        return inflater!!.inflate(R.layout.hero_fragment, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        text.setText("Hero item id " + getArguments().getLong(ARG_HERO_ID))
    }

    companion object {
        public val ARG_HERO_ID: String = "hero_id"
    }
}
